File: personalfinances.client/src/app/components/core/home/transaction.model.ts
export interface Transaction {
  stampEntity: string;           // Identificador único (gerado pelo backend)
  description: string;           // Descrição da transacção
  amount: number;                // Valor da transacção
  date: string;                  // Data (formatada como 'yyyy-MM-dd')
  category: 'income' | 'expense';// Categoria: "income" ou "expense"
  paymentMethod: 'cash' | 'credit_card' | 'debit_card' | 'bank_transfer'; // Método de pagamento
  recipient: string;             // Destinatário da transacção
  status: 'pending' | 'confirmed' | 'cancelled'; // Estado da transacção
  created_at?: string;           // Data de criação (opcional, se pretender exibi-la)
  updated_at?: string;           // Data de actualização (opcional)
}

File: PersonalFinances.BLL/Entities/Models/TransactionModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PersonalFinances.BLL.Entities.Models
{
    public class TransactionModel : BaseEntity
    {
        public string Description { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public string Category { get; set; } // "income" ou "expense"
        public string PaymentMethod { get; set; } // "cash", "credit_card", "debit_card", "bank_transfer"
        public string Recipient { get; set; }
        public string Status { get; set; } // "pending", "confirmed", "cancelled"
    }

}
